app/policies/group_offer_policy.rb

```diff
@@ -0,0 +1,16 @@
+class GroupOfferPolicy < ApplicationPolicy
+  class Scope < ApplicationScope
+    def resolve
+      return all if superadmin?
+      resolve_department if department_manager?
```

Comment:
Not sure if it's a problem if `Scope#resolve` returns `nil`, alternatively you can use [`none`](http://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-none) to return a scope which will never return any records.

Reply:
This triggers some additional unnecessary queries, why not simply:

```ruby
if superadmin?
  all
elsif department_manager?
  resolve_department
else
  none
end
```

(and also add `delegate :none, to: :scope` in `ApplicationScope`)